java-plugin/src/main/java/fr/greencodeinitiative/java/checks/CookieWithoutExpirationRule.java

```diff
@@ -0,0 +1,100 @@
+package fr.greencodeinitiative.java.checks;
+import java.util.*;
+
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.declaration.VariableTreeImpl;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.*;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+
+@Rule(
+        key = "CRJVM207-JAVA",
+        name = "Developpement",
+        description = CookieWithoutExpirationRule.MESSAGERULE,
+        priority = Priority.MINOR,
+        tags = {"smell"})
+
+public class CookieWithoutExpirationRule extends IssuableSubscriptionVisitor {
+
+    protected static final String MESSAGERULE = "Customer data must have end-of-life information, so cookies must have a maxAge";
+
+
+ @Override
+ public List<Tree.Kind> nodesToVisit() {
+     return Collections.singletonList(Tree.Kind.CLASS);
+ }
+ private final CookieWithoutExpirationRuleCheckVisitor visitorInFile = new CookieWithoutExpirationRuleCheckVisitor();
+
+
+ @Override
+ public void visitNode(Tree tree) {
+
+     tree.accept(visitorInFile);
+     //Class visitor
+     if (visitorInFile.hasANewCookieWithoutMaxDate())
+     {
+         //if we found a cookie that maxDate is not initialized, we report issue
+         reportIssue(tree, "Avoid not setting MaxAge");
```

Comment:
message with a double negative is difficult to understand : please use a simple message like `Giving a MaxAge is recommended`